lib/tests/test_gpg.rs

```diff
@@ -218,3 +379,124 @@ fn invalid_signature() {
+        "8F08B1D7DDEEFA91C25F8681F6494FF2E03DEADE"
+    );
+    assert_eq!(check.display.unwrap(), "/CN=JJ/O=X509 Signing Test");
+
+    let check = backend.verify(b"so so bad", &signature).unwrap();
+    assert_eq!(check.status, SigStatus::Bad);
+    assert_eq!(
+        check.key.unwrap(),
+        "8F08B1D7DDEEFA91C25F8681F6494FF2E03DEADE"
+    );
+    assert_eq!(check.display.unwrap(), "/CN=JJ/O=X509 Signing Test");
+}
+
+#[test]
+#[cfg_attr(windows, ignore = "stuck randomly on Windows CI #3140")] // FIXME
+fn gpgsm_signing_roundtrip_explicit_key() {
+    gpgsm_guard!();
+
+    let env = GpgsmEnvironment::new().unwrap();
+    let backend = gpgsm_backend(&env);
+    let data = b"hello world";
+    let signature = backend.sign(data, Some("someone@example.com")).unwrap();
+
+    assert_debug_snapshot!(backend.verify(data, &signature).unwrap(), @r#"
+    Verification {
+        status: Good,
+        key: Some(
+            "8F08B1D7DDEEFA91C25F8681F6494FF2E03DEADE",
+        ),
+        display: Some(
+            "/CN=JJ/O=X509 Signing Test",
+        ),
+    }
+    "#);
+    assert_debug_snapshot!(backend.verify(b"so so bad", &signature).unwrap(), @r#"
+    Verification {
+        status: Bad,
+        key: Some(
+            "8F08B1D7DDEEFA91C25F8681F6494FF2E03DEADE",
+        ),
+        display: Some(
+            "/CN=JJ/O=X509 Signing Test",
+        ),
+    }
+    "#);
+}
+
+#[test]
+#[cfg_attr(windows, ignore = "stuck randomly on Windows CI #3140")] // FIXME
+fn gpgsm_unknown_key() {
+    gpgsm_guard!();
+
+    let env = GpgsmEnvironment::new().unwrap();
+    let backend = gpgsm_backend(&env);
+    let signature = br"-----BEGIN SIGNED MESSAGE-----
+    MIAGCSqGSIb3DQEHAqCAMIACAQExDzANBglghkgBZQMEAgEFADCABgkqhkiG9w0B
+    BwEAADGCAnYwggJyAgEBMDUwKTEaMBgGA1UEChMRWDUwOSBTaWduaW5nIFRlc3Qx
+    CzAJBgNVBAMTAkpKAgh8bds9GXiZmzANBglghkgBZQMEAgEFAKCBkzAYBgkqhkiG
+    9w0BCQMxCwYJKoZIhvcNAQcBMBwGCSqGSIb3DQEJBTEPFw0yNTAzMTgyMDAzNDBa
+    MCgGCSqGSIb3DQEJDzEbMBkwCwYJYIZIAWUDBAECMAoGCCqGSIb3DQMHMC8GCSqG
+    SIb3DQEJBDEiBCCpSJBPLw9Hm4+Bl2lLMBhLDS7Rwc0qHsD7hdKZoZKkRzANBgkq
+    hkiG9w0BAQEFAASCAYANOvWCJuOKn018s731TWFHq5wS13xB7L83/2q8Mi9cQ3YT
+    kq8CQlyJV0spIW7dwztjsllX8X2szE4N0l83ghf3ol6B6n9Vyb844oKgb6cwc9uX
+    S8D1yiaj1Mfft3PDp+THH+ESezw1Djzj7E53Yx5j3kna/ylJhheg3raWit2MUxI0
+    V42Svm4PLcpOf+ywzstlSSx9p6Y8woctdkMkpyivNCsfwlRARFGSTP3G9DXZNv03
+    WZ51zlMT8lsYbT9EJUxzXuEpcJZJL0TYcbJ3n7uSopivHk843onIc71gbH/ByuMp
+    qokJ7jYzEMrk0YowzsD7wrtwhF5OgpW5ane8vuyquLOrRNX9H/TooE4+8OCM6nvQ
+    w7jgv1/hsdtDnZCkVaM0plhb2btE7Awgol5M8f9IDz1Z+b0t4ydc/iqHtE9yaqvZ
+    +aT9XXKKcj9XBhi1S790B4r8YoDyeiyzBs0gwvMuWjWMS7wixTbgx+IkQUrkgTLY
+    xiNbRmGtEonl9d8JS/IAAAAAAAA=
+    -----END SIGNED MESSAGE-----
+    ";
+    assert_matches!(
+        backend.verify(b"hello world", signature),
+        Err(SignError::InvalidSignatureFormat)
+    );
+    assert_matches!(
+        backend.verify(b"so bad", signature),
+        Err(SignError::InvalidSignatureFormat)
+    );
```

Comment:
nit: can you leave a comment saying that this is actually no certificates found error or something like that?

As far as I understand it, `gpgsm` doesn't have `NO_PUBKEY` status code or equivalent. We'll probably need to rename the error `InvalidSignatureFormat` because there are other kind of errors mapped to this.